### ISSUE TITLE
[tune/rllib] Better Usability for restoring from last checkpoint

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#7927]
### Describe your feature request

Restoring from last checkpoint requires a bit of wrangling (`max(glob.glob)`). It would be nice if a user could pass in a directory, and we look for the most recent checkpoint and restore directly from that directory.

**Comment by ttumiel** [comment]
Isn't this done by the `restore` argument in ray.tune?

**Comment by stefanbschneider** [comment]
I have implemented helper functions for loading the last checkpoint already and would be happy to contribute them. I'll check to find a suitable place in the code and open a PR.

It'd be my first ray PR (other than adding documentation). Anything specific to keep in mind?

**Comment by richardliaw** [comment]
go ahead and open it!

**Comment by stefanbschneider** [comment]
Ok, I opened a PR here: https://github.com/ray-project/ray/pull/14735

I tried to follow all contribution guidelines. Tests are failing, but I don't think it's due to my changes. Seems like my tests are passing.

**Comment by stefanbschneider** [comment]
I am currently still debating what we want a `get_last_checkpoint()` function to do exactly. Load the last checkpoint in terms of creation time or in terms of number of training steps? 

If there are multiple trials in the directory (eg, belonging to different hyperparameters), which trial should be loaded (by default)? The last one or the best one (in terms of some metric)? Or throw an error if there are multiple trials and no particular trial was specified?
In my use cases, I always only had a single trial anyways and didn't want to bother with specifying any particular trial, so I'd make that optional.
